From d7f482afd4138a8f1cd0b161ac2a3f37f9757c69 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Thu, 6 Aug 2020 11:03:03 +0800
Subject: [PATCH] policy/modules/system/systemd: allow systemd_generator_t to
 run systemd-getty-generator and systemd-fstab-generator

Fixes:
avc:  denied  { setfscreate } for  pid=125 comm="systemd-getty-g"
scontext=system_u:system_r:systemd_generator_t:s0-s15:c0.c1023
tcontext=system_u:system_r:systemd_generator_t:s0-s15:c0.c1023
tclass=process permissive=0

avc:  denied  { dac_override } for  pid=124 comm="systemd-fstab-g"
capability=1
scontext=system_u:system_r:systemd_generator_t:s0-s15:c0.c1023
tcontext=system_u:system_r:systemd_generator_t:s0-s15:c0.c1023
tclass=capability permissive=0

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/systemd.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index 52a0f77cd..4b8b325d7 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -365,6 +365,8 @@ seutil_search_default_contexts(systemd_coredump_t)
 #
 
 allow systemd_generator_t self:fifo_file rw_fifo_file_perms;
+allow systemd_generator_t self:capability dac_override;
+allow systemd_generator_t self:process setfscreate;
 
 corecmd_getattr_bin_files(systemd_generator_t)
 
-- 
2.17.1

